src/reactionsystem.jl

```diff
@@ -1221,7 +1222,34 @@ function set_default_metadata(rs::ReactionSystem;  default_reaction_metadata = [
     updated_equations = map(eqtransform, get_eqs(rs))
     @set! rs.eqs = updated_equations
     @set! rs.rxs = Reaction[rx for rx in updated_equations if rx isa Reaction]
-    
+
+    # Special routine to handle `Reaction` metadata that can contain new symbolic variables.
+    # Currently, `noise_scaling` is the only relevant metadata supported this way.
+    drm_dict = Dict(default_reaction_metadata)
+    if haskey(drm_dict, :noise_scaling)
+        # Finds parameters, species, and variables in the noise scaling term.
+        ns_expr = drm_dict[:noise_scaling]
+        ns_syms = [Symbolics.unwrap(sym) for sym in get_variables(ns_expr)]
+        ns_ps = filter(ModelingToolkit.isparameter, ns_syms)
+        ns_sps = filter(Catalyst.isspecies, ns_syms)
+        ns_vs = filter(sym -> !Catalyst.isspecies(sym) && !ModelingToolkit.isparameter(sym), ns_syms)
+
+        # Adds parameters, species, and variables to the `ReactionSystem`.
+        if any(!any(isequal(p1, p2) for p2 in get_ps(rs)) for p1 in ns_ps)
+            @set! rs.ps = unique([get_ps(rs); ns_ps])
+        end
+        if any(!any(isequal(sp1, sp2) for sp2 in get_species(rs)) for sp1 in ns_sps)
+            sps_new = unique([get_species(rs); ns_sps])
+            vs_old = get_unknowns(rs)[length(get_species(rs))+1 : end]
+            @set! rs.species = sps_new
+            @set! rs.unknowns = [sps_new; vs_old]
+        end
+        if any(!any(isequal(v1, v2) for v2 in get_unknowns(rs)) for v1 in ns_vs)
+            us_new = unique([get_unknowns(rs); ns_vs])
+            @set! rs.unknowns = us_new
+        end
+    end
```

Comment:
Does `union!` not work?

Reply:
I never really knew about `union!` until I discovered that you use it in a few cases, so not that familiar with it. That said I should start using it, will implement it here.

Reply:
Yeah, I can probably use generators, I think at this stage it was late and I was getting increasingly annoyed at all the effort doing this extra step. Will update to use generators though.

Reply:
actually, I am a little bit uncertain exactly how you suggest we use `union!` and generators in these cases